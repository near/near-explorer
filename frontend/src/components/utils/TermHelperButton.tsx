import { Row } from "react-bootstrap";
import React from "react";

import Modal from "./Modal";

interface Props {
  title: string;
}

interface State {
  modalInfo?: string;
}

export default class extends React.Component<Props, State> {
  state: State = {};

  changeModalHandler = (name: string) => {
    this.setState({ modalInfo: name });
  };

  ModalCloseHander = () => {
    this.setState({ modalInfo: undefined });
  };

  render() {
    return (
      <Row className="term-helper" noGutters>
        <img
          src="/static/images/icon-info.svg"
          className="info"
          onClick={() => this.changeModalHandler(this.props.title)}
        />
        {this.state.modalInfo && (
          <Modal modalClosed={this.ModalCloseHander}>
            {this.props.children}
          </Modal>
        )}
        <style jsx global>{`
          .term-helper {
            display: inline-block;
            width: 14px;
            height: 14px;
            text-transform: none;
          }

          .info {
            display: none;
            vertical-align: text-bottom;
            margin-left: 5px;
            width: 16px;
            cursor: pointer;
          }

          .term-helper:hover .info {
            display: block;
          }

          @media (max-width: 800px) {
            .info {
              display: block;
            }
          }
        `}</style>
      </Row>
    );
  }
}
