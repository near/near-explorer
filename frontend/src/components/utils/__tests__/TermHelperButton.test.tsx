import renderer from "react-test-renderer";

import TermHelperButton from "../TermHelperButton";

describe("<CardCell />", () => {
  it("renders", () => {
    expect(
      renderer.create(
        <TermHelperButton title={"Nodes Online"}>
          {
            "The number of validating nodes / the total number of online nodes. "
          }
          <a
            href={
              "https://docs.nearprotocol.com/docs/roles/integrator/faq#validators"
            }
          >
            docs
          </a>
        </TermHelperButton>
      )
    ).toMatchSnapshot();
  });
});
