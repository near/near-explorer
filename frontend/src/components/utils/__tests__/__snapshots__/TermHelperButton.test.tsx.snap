// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCell /> renders 1`] = `
<div
  className="term-helper row no-gutters"
>
  <img
    className="jsx-652848287 info"
    onClick={[Function]}
    src="/static/images/icon-info.svg"
  />
</div>
`;
